src/ryu.jl

```diff
@@ -41,39 +41,70 @@ function format_fixed_scientific(x::AbstractFloat, precision::Integer,
     end
 
     if engineering
-        b, e10 = Ryu.reduce_shortest(convert(Float32, x))
-        d, r = divrem(e10, 3)
-        if d < 0 &&
-            d += sign(r)
-
+        base_digits, power = get_engineering_string(x, precision)
+    else
+        e_format_number = Ryu.writeexp(x, precision)
+        base_digits, power = split(e_format_number, 'e')
     end
-    ryustr = Ryu.writeexp(x, precision)
-    @show x ryustr
 
-    # Rewrite the exponent
+
     buf = IOBuffer()
-    ret = iterate(ryustr)
-    while ret !== nothing
-        c, state = ret
-        c === 'e' && break
-        print(buf, c)
-        ret = iterate(ryustr, state)
+
+    print(buf, base_digits)
+    print(buf, "×10")
+
+    if power[1] == '-'
+        print(buf, '⁻')
+    end
+    leading_index = findfirst(c -> '1' <= c <= '9', power)
```

Comment:
need to find the first non-zero digit, as ryu outputs double digit exponents = `1.12e3 = 1.12e+03`